patients/summary.go

```diff
@@ -0,0 +1,155 @@
+	TimeInVeryHighPercent           *float64 `bson:"timeInVeryHighPercent,omitempty" json:"timeInVeryHighPercent,omitempty"`
+	TimeInVeryHighPercentDelta      *float64 `bson:"timeInVeryHighPercentDelta,omitempty" json:"timeInVeryHighPercentDelta,omitempty"`
+	TimeInVeryHighRecords           *int     `bson:"timeInVeryHighRecords,omitempty" json:"timeInVeryHighRecords,omitempty"`
+	TimeInVeryHighRecordsDelta      *int     `bson:"timeInVeryHighRecordsDelta,omitempty" json:"timeInVeryHighRecordsDelta,omitempty"`
+	TimeInVeryLowMinutes            *int     `bson:"timeInVeryLowMinutes,omitempty" json:"timeInVeryLowMinutes,omitempty"`
+	TimeInVeryLowMinutesDelta       *int     `bson:"timeInVeryLowMinutesDelta,omitempty" json:"timeInVeryLowMinutesDelta,omitempty"`
+	TimeInVeryLowPercent            *float64 `bson:"timeInVeryLowPercent,omitempty" json:"timeInVeryLowPercent,omitempty"`
+	TimeInVeryLowPercentDelta       *float64 `bson:"timeInVeryLowPercentDelta,omitempty" json:"timeInVeryLowPercentDelta,omitempty"`
+	TimeInVeryLowRecords            *int     `bson:"timeInVeryLowRecords,omitempty" json:"timeInVeryLowRecords,omitempty"`
+	TimeInVeryLowRecordsDelta       *int     `bson:"timeInVeryLowRecordsDelta,omitempty" json:"timeInVeryLowRecordsDelta,omitempty"`
+	TotalRecords                    *int     `bson:"totalRecords,omitempty" json:"totalRecords,omitempty"`
+	TotalRecordsDelta               *int     `bson:"totalRecordsDelta,omitempty" json:"totalRecordsDelta,omitempty"`
+}
+
+type PatientCGMPeriods map[string]PatientCGMPeriod
+
+type PatientCGMStats struct {
+	Config        *PatientSummaryConfig `bson:"config,omitempty" json:"config,omitempty"`
+	Dates         *PatientSummaryDates  `bson:"dates,omitempty" json:"dates,omitempty"`
+	OffsetPeriods *PatientCGMPeriods    `bson:"offsetPeriods,omitempty" json:"offsetPeriods,omitempty"`
+	Periods       *PatientCGMPeriods    `bson:"periods,omitempty" json:"periods,omitempty"`
+	TotalHours    *int                  `bson:"totalHours,omitempty" json:"totalHours,omitempty"`
+}
+
+type PatientSummaryConfig struct {
+	HighGlucoseThreshold     *float64 `bson:"highGlucoseThreshold,omitempty" json:"highGlucoseThreshold,omitempty"`
+	LowGlucoseThreshold      *float64 `bson:"lowGlucoseThreshold,omitempty" json:"lowGlucoseThreshold,omitempty"`
+	SchemaVersion            *int     `bson:"schemaVersion,omitempty" json:"schemaVersion,omitempty"`
+	VeryHighGlucoseThreshold *float64 `bson:"veryHighGlucoseThreshold,omitempty" json:"veryHighGlucoseThreshold,omitempty"`
+	VeryLowGlucoseThreshold  *float64 `bson:"veryLowGlucoseThreshold,omitempty" json:"veryLowGlucoseThreshold,omitempty"`
+}
+
+// PatientSummaryDates dates tracked for summary calculation
+type PatientSummaryDates struct {
+	FirstData         *time.Time `bson:"firstData,omitempty" json:"firstData,omitempty"`
+	HasFirstData      *bool      `bson:"hasFirstData,omitempty" json:"hasFirstData,omitempty"`
```

Comment:
More for my understanding but hasFirstData is equivalent to firstData not being empty?

Reply:
correct.